haystack/nodes/prompt/invocation_layer/sagemaker.py

```diff
@@ -0,0 +1,303 @@
+import json
+import logging
+from typing import Optional, Dict, Union, List, Any
+
+import requests
+
+from haystack.errors import SageMakerModelNotReadyError, SageMakerInferenceError, SageMakerConfigurationError
+from haystack.lazy_imports import LazyImport
+from haystack.nodes.prompt.invocation_layer import PromptModelInvocationLayer
+from haystack.nodes.prompt.invocation_layer.handlers import DefaultPromptHandler
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install farm-haystack[aws]'") as boto3_import:
+    import boto3
+    from botocore.exceptions import ClientError, BotoCoreError
+
+
+class SageMakerInvocationLayer(PromptModelInvocationLayer):
+    """
+    SageMaker Invocation Layer
+
+    Lets you use an LLM hosted via a SageMaker Inference Endpoint PromptNode.
+    It's expecting a Huggingface Model running at the SageMaker Inference Endpoint (e.g. Falcon, Flan-T5).
+    See https://www.philschmid.de/sagemaker-falcon-llm for instructions on how to deploy such a model to SageMaker
+    and https://docs.aws.amazon.com/sagemaker/latest/dg/jumpstart-foundation-models-use.html for an even simpler
+    workflow via SageMaker JumpStart.
+
+    **Example**
+
+     ```python
+    from haystack.nodes import PromptNode
+
+    # Pass sagemaker endpoint name and authentication details
+    pn = PromptNode(model_name_or_path="falcon-40b-my-sagemaker-inference-endpoint,
```

Comment:
You are missing a '"'

`pn = PromptNode(model_name_or_path="falcon-40b-my-sagemaker-inference-endpoint",`

Reply:
You are right. I'll remember this one when we touch the file for the first time 👍